generator/testdata/rust/gclient/golden/secretmanager/src/lib.rs

```diff
@@ -419,3 +419,88 @@ impl SecretManagerServiceClient {
         Ok(response)
     }
 }
+
+/// Manages location-related information with an API service.
+#[derive(Clone)]
+pub struct LocationsClient {
+    inner: Arc<InnerClient>,
+}
+
+impl LocationsClient {
+    pub async fn new() -> Result<Self> {
+        Self::new_with_config(ConfigBuilder::new()).await
+    }
+
+    pub async fn new_with_config(conf: ConfigBuilder) -> Result<Self> {
+        let inner = InnerClient {
+            http_client: conf.client.unwrap_or(ConfigBuilder::default_client()),
+            cred: conf
+                .cred
+                .unwrap_or(ConfigBuilder::default_credential().await?),
+            endpoint: conf.endpoint.unwrap_or(DEFAULT_HOST.to_string()),
+        };
+        Ok(Self {
+            inner: Arc::new(inner),
+        })
+    }
+
+    /// Lists information about the supported locations for this service.
+    pub async fn list_locations(
```

Comment:
I think we should include the integration test changes in this PR, but I can easily be convinced we can go a different way.

Reply:
That is fair, will add some.